packages/core/src/bind/index.ts

```diff
@@ -46,12 +46,18 @@ export function bind<T>(
  * subscription, then the hook will leverage React Suspense while it's waiting
  * for the first value.
  */
-export function bind<A extends (number | string | boolean | null)[], O>(
+export function bind<
+  A extends (number | string | boolean | null | Object | Symbol | undefined)[],
+  O
+>(
```

Comment:
Do we need to specific what types it supports? Could it be `unknown[]` or `any[]` instead?